pandas/tests/tools/test_merge.py

```diff
@@ -1355,3 +1355,51 @@ def test_dtype_on_merged_different(self, change, how, left, right):
                            np.dtype('int64')],
                           index=['X', 'Y', 'Z'])
         assert_series_equal(result, expected)
+
+
+@pytest.fixture
+def df1():
```

Comment:
I have also changed the order of the right index (like in pandas/tests/frame/test_join.py), so that sort=False is tested for the right join.